lib/commands/backup.js

```diff
@@ -0,0 +1,230 @@
+                if (err.code === 'EISDIR')  { // Symlink
+                    return this.backupContent(ctx, null, localFile).catch((error) => {
+                        debug(error);
+                        ctx.instance.ui.log(`Failed to backup ${localFile}`, 'yellow');
+                        // Even though this specific file couldn't be backed up, we want to continue
+                        return Promise.resolve();
+                    });
+                } else {
+                    debug(err);
+                    ctx.instance.ui.log(`Failed to backup ${localFile}`, 'yellow');
+                    // Even though this specific file couldn't be backed up, we want to continue
+                    return Promise.resolve();
+                }
+            });
+        });
+
+        return Promise.all(promises);
+    }
+
+    backupConfig(ctx) {
+        debug('Intializing zip file');
+        try {
+            const configFileName = `config.${ctx.system.environment}.json`;
+            const readFile = fs.readFileSync;
+
+            // Add environment-specific config file
+            ctx.zipFile.addFile(configFileName, readFile(configFileName), '', 644);
+            // Add CLI config file
+            ctx.zipFile.addFile('.ghost-cli', readFile('.ghost-cli'), '', 644);
+
+            return Promise.resolve();
+        } catch (error) {
+            debug(`Failed creating zip: ${error}`);
+            return Promise.reject(new errors.SystemError({
+                context: error,
+                message: 'Failed backing up configuration files'
+            }));
+        }
+    }
+
+    backupExtensions(ctx) {
+        const isArray = require('lodash/isArray');
+        const read = fs.readFileSync;
+
+        return ctx.system.hook('backup').then((extensionFiles) => {
+            extensionFiles.forEach((filesToAdd) => {
+                if (isArray(filesToAdd)) {
+                    filesToAdd.forEach((metadata) => {
+                        try {
+                            ctx.zipFile.addFile(`extension/${metadata.fileName}`, read(metadata.location), '', 644);
+                        } catch (error) {
+                            ctx.ui.log(`Not backing up "${metadata.description || metadata.location}" - ${error.message}`);
+                        }
+                    });
+                } /* else if(filesToAdd !== undefined) {
+                    // @todo: do we need to do anything here?
+                    debug('Bad data passed');
+                }*/
+            });
+        }).catch(Promise.reject);
+    }
+
+    writeZipFile(ctx) {
+        // Write zip file to specified folder
+        debug(`Writing zipFile to ${ctx.saveLocation}`);
+        ctx.zipFile.writeZip(ctx.saveLocation);
+        ctx.instance.ui.log(`Wrote backup to ${ctx.saveLocation}`);
+        // clear up memory
+        ctx.zipFile = null;
+    }
+
+    backupDatabase(ctx) {
+        debug('Creating database backup');
+        let exporter;
```

Comment:
ok, so I think I found a way to work around the fact that the database connection isn't terminated by the exporter.

```js
let exporter, db;
try {
  // This is based on cwd, not cli-install location
  db = require(absolutePath('current/core/server/data/db/'));
  exporter = require(absolutePath('current/core/server/data/export/'));
} catch (error) {
  // code omitted for brevity
}
return exporter.doExport().then((database) => {
  // code omitted
  return Promise.fromCallback((cb) => db.knex.destroy(cb));
}); // rest of code omitted for brevity
```

Did a quick test of the above code locally and it worked :)